tez-mapreduce/src/main/java/org/apache/tez/mapreduce/hadoop/MRInputHelpers.java

```diff
@@ -889,4 +895,29 @@ public static int getDagAttemptNumber(Configuration conf) {
     return getIntProperty(conf, MRInput.TEZ_MAPREDUCE_DAG_ATTEMPT_NUMBER);
   }
 
+  public static MRSplitProto getProto(InputDataInformationEvent initEvent, JobConf jobConf) throws IOException {
+    return Strings.isNullOrEmpty(initEvent.getSerializedPath()) ? readProtoFromPayload(initEvent)
+      : readProtoFromFs(initEvent, jobConf);
+  }
+
+  private static MRSplitProto readProtoFromFs(InputDataInformationEvent initEvent, JobConf jobConf) throws IOException {
+    String serializedPath = initEvent.getSerializedPath();
+    Path filePath = new Path(serializedPath);
+    LOG.info("Reading InputDataInformationEvent from path: {}", filePath);
+
+    MRSplitProto splitProto = null;
+    FileSystem fs = FileSystem.get(filePath.toUri(), jobConf);
+
+    try (FSDataInputStream in = fs.open(filePath)) {
+      splitProto = MRSplitProto.parseFrom(in);
+      fs.delete(filePath, false);
```

Comment:
If the parsing fails, in that case the file won't be deleted. is that expected? and if your delete fails our operation will fail as well, so it that ok or we can live with that?

Reply:
In my understanding yes, this is expected.

Reply:
yeah, I think under normal circumstances everything should work: parse + delete
any of them failing is fine to lead to a task attempt failure (and eventually rerun)
regarding leftover files: it's okay, current hive patch uses the tez scratch dir, which is supposed to be cleaned up by hive as well